api/src/main/java/io/opentelemetry/tags/unsafe/TagMapInScope.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.tags.unsafe;
+
+import io.grpc.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.tags.TagMap;
+
+/**
+ * A scope that manages the {@link Context} for a {@link TagMap}.
+ *
+ * @since 0.1.0
+ */
+public final class TagMapInScope implements Scope {
```

Comment:
Instead of exposing this class can we expose a method on ContextUtils:

```java
public Scope withTagMap(TagMap tagMap) {
...
}
```

Any thoughts on this approach?

Reply:
That sounds good - that being said, I wished we had something more general-purpose, like `<T> Scope withValue(T value);`

Reply:
It is not possible to have this, because we need access to the Key in this module as well.